### ISSUE TITLE
[java] NullAssignment - FP with ternary and null as method argument

### ISSUE THREAD

**Issue by KroArtem** [pmd/pmd#2320]
**Affects PMD Version:** 
6.20

**Rule:** [NullAssignment](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#nullassignment)
**Description:**

**Code Sample demonstrating the issue:**

```java
public String methodOne(String one, String two) {
        return "test";
    }
public String methodTwo(String one, String two) {
        List<String> strings = new ArrayList<>();
->>        return methodOne("one", strings.isEmpty() ? null : "two");
    }
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Programmatically

**Comment by KroArtem** [comment]
Possible duplicate of https://github.com/pmd/pmd/issues/1050 , https://github.com/pmd/pmd/issues/1669

Will look for root of the problem later

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
